### DISCUSSION TITLE
Welcome to CIMTool Discussions!

### DISCUSSION THREAD

**Discussion by ucaiug-admin** [cimug-org/CIMTool#36] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by choompol-boonmee** [comment]
I am Choompol from Thailand.
I have been using CIMTool to build CIM profile for years. (currently using version 1.11)
I would like to learn more about CIMTool.
I could not find to more in-dept tutorials anywhere to benefit more from it.
I also want to understand deeper in software and extend some functionality.
And I am willing to help.
I appreciate any advice.

**Reply by tviegut** [reply]
@choompol-boonmee :  Thanks for reaching out!  First off, great to meet you in-person this week at the UCA Int'l CIM Users Group in Arnhem.  Glad you could make it all the way from Thailand to join us in the NL.  :)   Yes, you're highlighting an area that we do have a need to shore up.  As I presented at the User's Group we have been working to migrate content onto publicly available websites that are easily searchable to the CIM Community (and others interested in learning more about the CIM).  A few examples thus far:

- The official "CIM Modeling Guide" that contains the rules and processes for modelling and extending the CIM is now publicly available at [CIM Modeling Guide](https://cim-mg.ucaiug.io).  It is also "mobile friendly" with PDF versions of the site available for download.
- CIMTool's new website is now located at [CIMTool](https://cimtool.ucaiug.io).  Note that we have plans in the works to build out gaps in documentation and provide video-based training on various topics and features in CIMTool.  These available, in turn, from its primary site and the UCA user group YouTube channel.  
- CIMTool's companion site [CIMTool Builders](https://cimtool-builders.ucaiug.io)

We would welcome your contributions in building out documentation to support the gaps you've identified.  

Todd

**Comment by SanjayGautam** [comment]
Hi Guys , 

I have joined my modeling team in the US western coast ISO , we use this tool and I see that tool can generate json and xsds using the profile , I am looking forward to contribute in this open forum so that I can help in moving from SOAP based world to the Open Api world. 
I am working out of California .

Just wanted to say Hi to Every one. 

Thanks
Sanjay Gautam

**Reply by tviegut** [reply]
@SanjayGautam :  Hi Sanjay.  Great to meet you.  I'm the lead for the CIMTool project as well as the lead of the IEC62361-104 "CIM Profiles to JSON Schema Mapping" standard.  I worked with your team in early 2020 (if I recollect the timing right) on CIMTool and early questions they had.  I believe I had presented an early draft of the spectication for JSON which led to their using a reference implementation of it.  I think it may have been Arun from CAISO who now is at EPRI.  In any event, we're now at release 2.0.1 with an upcoming 2.1.0 in the near future.  If you haven't already done so I'd suggest adding yourselve as a watcher (minimally for releases) for CIMTool so you can stay up to date with bug fixes, etc.  Please note, the JSON schema builder that is packaged with CIMTool is based on a draft of the 104 standard and will be ultimately changed when the publication goes to the IEC for I.S.  We've nearly finished up a final review with the N.C representatives at which point we will be circulating it for CDV stage (Committee Draft for Vote).  What version of CIMTool are you currently using internally?  Refer to the release notes section of CIMTool's website (https://cimtool.ucaiug.io) for new features/bug fixes.

**Reply by SanjayGautam** [reply]
Thank you so much for responding. Yes, the person who worked with you was Varun. I am now on the same team, and I am ramping up my knowledge of the tool. The great thing is that I am a programmer as well as an electrical engineer, and I really want to contribute to these CIM projects. Please let me know how I can help you and how I can best contribute to the energy domain's transition into the new world of REST APIs and microservices. This will help the domain avoid obsolescence. There has been pressure to move to openAPI-based platforms like Swagger, and while I understand their limitations, it is too early for me to comment definitively. This month, I have checked out the code and am exploring how the CIMTOOL plugin works. Please reach out to me if you need any help. I will be more than happy to contribute.

**Reply by SanjayGautam** [reply]
The version we are using is [1.10.0.RC1](https://github.com/cimug-org/CIMTool/releases/tag/1.10.0.RC1). I am planning to use the latest version as well. The first area that interests me is the JSON builder, because we want to customize the output and map it to the standard JSON data types, using formats like date-time and so on. We aim to provide some configuration so that the JSON created from the OWL file becomes more customizable.  Let see how far I will be able to accomplish that , may be you guys have already took care of it .

**Reply by ucaiug-admin** [reply]
> The version we are using is [1.10.0.RC1](https://github.com/cimug-org/CIMTool/releases/tag/1.10.0.RC1). I am planning to use the latest version as well. The first area that interests me is the JSON builder, because we want to customize the output and map it to the standard JSON data types, using formats like date-time and so on. We aim to provide some configuration so that the JSON created from the OWL file becomes more customizable.  Let see how far I will be able to accomplish that , may be you guys have already took care of it .

If you'd like to chat together before jumping in let me know. I can help to point you in the right direction.  Hit me up on LinkedIn to connect.

**Comment by tviegut** [comment]
@SanjayGautam :  One other thing.  For a bit more advanced users you might be interested in the [CIMTool builders library](https://cimtool-builders.ucaiug.io).  This would open up a broad set of options on custom builders.  For example, I'm involved in a collaborative effort with ENTSO-E in Europe on a reference implementation for Apache Avro schema generation based on CIM profiles (for streaming data).  Google Proto Buffers is another candidate for this type of thing. The link I provided here has detailed explanations and a variety of different types of builders that can be used as starting points.  If you do write a custom builder we strongly encourage submitting it back to the CIM community (which is the ideal purpose of that site).   Alex Anderson, a Systems Engineer at PNNL just contributed the [cimantic-graphs.xsl](https://cimtool-builders.ucaiug.io/custom-builders/cimantic-graphs/cimantic-graphs.xsl) builder that produces a specialized python dataclass schema that is used as input into the [CIMantic Graphs](https://github.com/PNNL-CIM-Tools/CIM-Graph/tree/develop) open source library.  That library is used to create, parse, and edit CIM power system models using in-memory knowledge graphs.  It's just one example of how to extend CIMTool via builders for specific problem spaces or target technologies.

**Reply by SanjayGautam** [reply]
@tviegut  thanks for the guidance , I will start looking at the CIMtool Builder library and right now I will need some customisation to the json builders and I will not then happy to Submit my changes and ideas to the CIM Community

**Comment by GEDavidPopoola** [comment]
Hi team,
I am David an Automation Engineer based in Berlin. I am currently working on an Electrification project and been using Excel for IEC61850 Signals list.
I am currently looking for a system that allows collaboration with other colleagues and also some level of automation, something which records changes, limits errors etc.
Could you recommend any of inform me if this system can do same?
Thank you.

**Reply by Sveino** [reply]
Hi David,
I have been working a very long time to stop the work by using Excel for IEC 61850 Signal list.. The big goal is to have IEC CIM (61970) harmonised with 61850. Work has been started and there are changes on both standards series. We are now planning for CIM18 to update the Measurement model to be better align. So we are very interested in any input. We are hoping that CIMTool can also help us mapping between CIM and 61850.
